### PR TITLE
Infra view data bags details page

### PR DESCRIPTION
**Pull request by sachin-msys**
Signed-off-by: sachin-msys <sbacchav@msystechnologies.com>

### :nut_and_bolt: Description: What code changed, and why?
Introduced Data-Bags details page view.

### :chains: Related Resources
Fixes: #3734
### :+1: Definition of Done
Data Bags details page of infra view where we are display the data bags item list and its respective details.
### :athletic_shoe: How to Build and Test the Change

1. git fetch origin Sachin/infra-view-data-bags-details
2. build components/automate-ui
3. To add data https://github.com/chef/automate/blob/master/dev-docs/adding-data/adding_test_data.md#adding-data-to-infra-views
4. Go to Infrastructure tab >> Chef Servers its under Chef-Server feature flag.
5. Click on `Server list` >> `Organisation list` >> `Data Bags Tab` >> then click on any of the data bag list item

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
data-bag-details

### CODE REVIEW

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+      this.store.select(allDataBagDetails)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      filter(([getDataBagDetailsSt, _dataBagDetailsState]) =>
+        getDataBagDetailsSt === EntityStatus.loadingSuccess),
+      filter(([_getDataBagDetailsSt, dataBagDetailsState]) =>
+        !isNil(dataBagDetailsState)))
+      .subscribe(([_getDataBagDetailsSt, dataBagDetailsState]) => {
+        this.dataBagDetails = dataBagDetailsState;
+        if (this.dataBagDetails.length) {
```
Comment (sachin-msys) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.ts:73]: This code block for select first item from the list by default.

File: components/automate-ui/src/app/entities/data-bags/data-bag-details.reducer.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+export interface DataBagDetailsEntityState extends EntityState<DataBags> {
+  getAllStatus: EntityStatus;
+}
+
+const GET_ALL_STATUS = 'getAllStatus';
+
+export const dataBagDetailsEntityAdapter: EntityAdapter<DataBags> = createEntityAdapter<DataBags>({
+  selectId: (dataBags: DataBags) => dataBags.name
+});
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/entities/data-bags/data-bag-details.reducer.ts:15]: this is a cool trick! 👍

File: components/automate-ui/src/app/entities/data-bags/data-bag-item-details.actions.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+export enum DataBagItemDetailsActionTypes {
+  GET = 'DATA_BAG_ITEM_DETAILS::GET',
+  GET_SUCCESS = 'DATA_BAG_ITEM_DETAILS::GET::SUCCESS',
+  GET_FAILURE = 'DATA_BAG_ITEM_DETAILS::GET::FAILURE'
+}
+
+// export interface DataBagItemDetailsSuccessPayload {
+//   data_bags_item_details: DataBagsItemDetails;
+// }
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/entities/data-bags/data-bag-item-details.actions.ts:13]: leftover comments

File: components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.spec.ts [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  MockComponent({ selector: 'chef-breadcrumbs' }),
+  MockComponent({ selector: 'chef-page-header' }),
+  MockComponent({ selector: 'chef-heading' }),
+  MockComponent({ selector: 'chef-subheading' }),
+  MockComponent({ selector: 'chef-loading-spinner' }),
+  DataBagsDetailsComponent
+];
+const serverId = '6e98f609-586d-4816-a6de-e841e659b11d';
+const orgId = '6e98f609-586d-4816-a6de';
+const dataBagName = 'demo_data_bag';
```
Comment (SEAjamieD) [inline components/automate-ui/src/app/modules/infra-proxy/data-bags-details/data-bags-details.component.spec.ts:36]: should these still be hardcoded?

**Comment by SEAjamieD** [comment]
has the script for loading data changed again recently?  i end up getting an error when I run it:
```
ERROR:
  Code: InvalidArgument
  Message: must supply org admin key
```

**Comment by vjeffrey** [comment]
> has the script for loading data changed again recently

yes, it did change pretty recently. the one pinned in chat (automate-infra-views) is uptodate

**Comment by SEAjamieD** [comment]
> > has the script for loading data changed again recently
> 
> yes, it did change pretty recently. the one pinned in chat (automate-infra-views) is uptodate

weird thought i tried that one 🤷‍♂️ - thanks! that worked perfect 👍

**Comment by SEAjamieD** [comment]
I noticed that when the Data Bag has no data associated, we just show 2 empty rows.  This may be a UX question, but is there a "no data" view?

Screen Shot 2020-06-09 at 4 37 34 PM

File: components/automate-ui/src/app/app-routing.module.ts [hunk 5]
```diff
@@ -262,6 +263,10 @@ const routes: Routes = [
             {
               path: ':id/organizations/:orgid/environments/:name',
               component: EnvironmentDetailsComponent
+            },
+            {
+              path: ':id/organizations/:orgid/data_bags/:name',
```
[the remark this answers is not in the document]
Comment (sachin-msys) [reply components/automate-ui/src/app/app-routing.module.ts:268]: @msorens After this changes there has to be good regression testing so created a separate issue ( #3924 ) to address this inconsistency.

File: components/automate-ui/src/app/modules/infra-proxy/org-details/org-details.component.ts [hunk 6]
```diff
@@ -52,20 +52,40 @@ export class OrgDetailsComponent implements OnInit, OnDestroy {
           this.environmentsTab = false;
```
[the remark this answers is not in the document]
Comment (sachin-msys) [reply components/automate-ui/src/app/modules/infra-proxy/org-details/org-details.component.ts]: thanks for this nice suggestion :+1:

**Comment by sachin-msys** [comment]
> > > I noticed that when the Data Bag has no data associated, we just show 2 empty rows. This may be a UX question, but is there a "no data" view?
> > > Screen Shot 2020-06-09 at 4 37 34 PM
> > 
> > 
> > @jonong1972 Need your suggestion here. We can show empty section with message if no data bag details available in that case need a message for the same.
> 
> I would follow the current pattern we are using where we show no table and have a line of text. Other areas have a call to action to add or create the missing data. Since there is no edit capability, I would just put a sentence to the extent of:
> 
> "No data available"
> 
> @susanev or others can chime in for more word-smithing, but I am good with what I suggested.
> 
> For reference - this is our current pattern
> 
> Screen Shot 2020-06-10 at 11 57 28 AM

@jonong1972 `No data available` suggestion of yours please have a look. 
no-data

**Comment by jonong1972** [comment]
Looks good on my end as well.

**Pull request merged**